File: Rasyonet_Intern.API/Entities/Category.cs
namespace Rasyonet_Intern.API.Entities
{
    public class Category
    {
        public int Id { get; set; }
        public string CategoryName { get; set; } = string.Empty;
        // One-to-many navigation property
        public List<Performance> Performances { get; set; } = new();
    }
}

File: Rasyonet_Intern.Mobile/types/category.ts
export type Category = {
    id: string
    categoryName: string
    performances: Performance[]
}
//Backend kısmında JSON yapısında tek tablo olarak yazdığımız için front end tarafında da tek tablo olarak yazıyoruz.
export type Performance = {
    uniqueCode: string
    performanceName: string
    value: number
    price: number
    dailyChange: number
    weeklyChange: number
    monthlyChange: number
}
// Backend'deki Document sınıflarının TypeScript karşılıklarıdır. Bu sayede API'den gelen verileri TypeScript tipleri ile eşleştirerek daha güvenli ve hatasız bir şekilde kullanabiliriz.